### PR TITLE
🦋 Update webhook edit page

### PR DESCRIPTION
**Pull request by josemigallas**
[THREESCALE-10268: Account Settings > Integrate > Webhooks](https://issues.redhat.com/browse/THREESCALE-10268)

Before:
Screenshot 2024-05-14 at 14 20 50

After:
Screenshot 2024-05-14 at 14 15 40

### CODE REVIEW

File: app/inputs/patternfly_checkbox_input.rb [hunk 1]
```diff
@@ -4,12 +4,14 @@ class PatternflyCheckboxInput < Formtastic::Inputs::BooleanInput
   delegate :tag, to: :template
 
   def to_html
+    checkbox = tag.div(class: 'pf-c-check') do
+      hidden_field_html + input + label + description
+    end
+
+    return checkbox if options[:nude]
```
Comment (mayorova) [inline app/inputs/patternfly_checkbox_input.rb:11]: Hmm... is that needed? I don't notice any changes if I drop this.

File: app/views/provider/admin/webhooks/edit.html.slim [hunk 2]
```diff
@@ -1,13 +1,75 @@
...
+        div class="pf-c-form__section-title" aria-hidden="true" Behavior
+        div class="pf-c-form__group"
+          = form.input :active, as: :patternfly_checkbox
+          = form.input :provider_actions, as: :patternfly_checkbox
+
+      section class="pf-c-form__section" role="group"
+        div class="pf-c-form__section-title" aria-hidden="true" Webhooks endpoint
+        div class="pf-c-form__group"
+          - show_action = @webhook.persisted? && @webhook.url.present?
+          = form.input :url, as: :patternfly_input, action:(show_action ? { title: 'Ping!', 'data-ping-url': provider_admin_webhooks_path } : nil)
```
Comment (mayorova) [inline app/views/provider/admin/webhooks/edit.html.slim:23]: It's kinda weird, that you need to actually enable webhooks, and set the URL, **and** save the form, and only then the Ping button will appear. But that was the previous behavior also, so it's fine.

File: app/lib/three_scale.rb [hunk 3]
```diff
@@ -24,6 +24,7 @@ def validate_settings
   end
 
   def master_on_premises?
+    # return true
```
Comment (mayorova) [inline app/lib/three_scale.rb:27]: :eyes:

File: features/provider/admin/webhooks/new.feature [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    When they select "Account Settings" from the context selector
+    And press "Integrate" within the main menu
+    And follow "Webhooks" within the main menu
+    Then the current page is the new webhook page
+
+  Scenario: Creating a webhook
+    Given they go to the new webhook page
+    And there shouldn't be a button to "Ping!"
+    When the form is submitted with:
+      | URL | http://google.com |
```
Comment (mayorova) [inline features/provider/admin/webhooks/new.feature:21]: I don't get why we are using `google.com` in our examples :upside_down_face: maybe it needs to be changed to something more redhatt-y or 3scale-ey

Comment (josemigallas) [reply features/provider/admin/webhooks/new.feature:21]: Ah, yes! In fact I meant to change it, but cucumbers sucked all my energy and I forgot. I will do it asap.

File: app/inputs/patternfly_checkbox_input.rb (hunk 1 above)
Comment (josemigallas) [reply app/inputs/patternfly_checkbox_input.rb]: I makes the resulting HTML exactly like Patternfly 4, but like you say, it doesn't affect either the layout or the tests so I will remove it.

**Review by mayorova — APPROVED** [review]
:+1:

**Pull request merged**